src/math/LeastSquares.cc

```diff
@@ -360,6 +360,7 @@ LeastSquares::LeastSquares(Factorization factorization, int dimension) {
             break;
     }
     _impl->factorization = factorization;
+    _impl->whichDiagnostic = factorization;
```

Comment:
Mind instead adding these as arguments to the `Impl` (base class) constructor, and having the derived constructors pass down their own constant?  This is really something the constructor should guarantee.